### PR TITLE
[prometheus-smartctl-exporter] Add support for device-include / device-exclude and image pull secrets

### PR DESCRIPTION
**Pull request by jmif**
#### What this PR does / why we need it

Adds image pull secret support and device regex support to smartctl monitor

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus-smartctl-exporter/Chart.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 0.11.0
+version: 0.11.1
```
Comment (zeritti) [inline charts/prometheus-smartctl-exporter/Chart.yaml:18]: The change should be a minor one:

```suggestion
version: 0.12.0
```

File: charts/prometheus-smartctl-exporter/templates/daemonset.yaml [hunk 2]
```diff
@@ -35,12 +35,24 @@ spec:
         {{- include "prometheus-smartctl-exporter.selectorLabels" $global | nindent 8 }}
         idx: i{{ $idx }}
     spec:
+      {{- if $global.Values.image.pullSecrets }}
+      imagePullSecrets:
```
Comment (zeritti) [inline charts/prometheus-smartctl-exporter/templates/daemonset.yaml:39]: Please, add the new field `image.pullSecrets` with its default value in `values.yaml`.

File: charts/prometheus-smartctl-exporter/templates/daemonset.yaml [hunk 3]
```diff
@@ -35,12 +35,24 @@ spec:
...
+        {{- end }}
+      {{- end }}
       containers:
       - image: "{{ $global.Values.image.repository }}:{{ $global.Values.image.tag | default $global.Chart.AppVersion }}"
         imagePullPolicy: {{ $global.Values.image.pullPolicy }}
         args:
         - '--smartctl.path={{ $config.smartctl_location }}'
         - '--smartctl.interval={{ $config.collect_not_more_than_period }}'
+{{- if $config.device_exclude }}
+        - '--smartctl.device-exclude={{ $config.device_exclude }}'
```
Comment (zeritti) [inline charts/prometheus-smartctl-exporter/templates/daemonset.yaml:51]: The new fields should be entered in `values.yaml` with their default values.

**Review by zeritti — CHANGES_REQUESTED** [review]
Thank you, @jmif, for your PR. Please, see my comments below.

File: charts/prometheus-smartctl-exporter/templates/daemonset.yaml (hunk 3 above)
Comment (jmif) [reply charts/prometheus-smartctl-exporter/templates/daemonset.yaml:51]: For the image pull secrets above I added an explicit default but I noticed here that the `config.devices` option does not have a default, so I followed the existing pattern in `values.yaml` and added commented out examples.  LMK if this is sufficient.

Comment (zeritti) [reply charts/prometheus-smartctl-exporter/templates/daemonset.yaml:51]: Thank you, @jmif. Please, sign your commit.

**Comment by jmif** [comment]
@zeritti done!

**Review by zeritti — APPROVED** [review]

**Pull request merged**